cli/js/testing.ts

```diff
@@ -18,47 +19,61 @@ export type TestFunction = () => void | Promise<void>;
 export interface TestDefinition {
   fn: TestFunction;
   name: string;
+  skip?: boolean;
 }
 
+type TestOptions = Omit<TestDefinition, "fn" | "name">;
+
 const TEST_REGISTRY: TestDefinition[] = [];
 
 export function test(t: TestDefinition): void;
-export function test(fn: TestFunction): void;
-export function test(name: string, fn: TestFunction): void;
+export function test(fn: TestFunction, options?: TestOptions): void;
+export function test(
+  name: string,
+  fn: TestFunction,
+  options?: TestOptions
+): void;
```

Comment:
Still iterating on testing API and I think for now, we should support `skip` only in `TestDefinition`; we need to be careful about overloads and I'd rather punt on adding new ones for a little longer.

Reply:
More than happy to remove it from the shorthands :) For the record, it only appended an options object to the existing overloads.